lib/jet/daemon.js

```diff
@@ -451,6 +459,7 @@ var createDaemon = function (options) {
       })
       daemon.wsServer.on('connection', function (ws) {
         var peer = peers.add(ws)
+        peerIdQueue.push(peer.id)
```

Comment:
I'm not sure why this is being done with a queue but either way, this is only done with Websocket peers. It should be independent of the connection method.

It would be better to the logic for sending the peer Id in peers.add(). At the end, before returning the peer, you can do peer.sendMessage({id: peer.id}). On the client, waiting for this message should just be part of connection process.